### PR TITLE
Add Creator union type

### PR DESCRIPTION
**Pull request by Narigo**
Sorry about the whitespace formatting. If you want to keep the whitespaces, I need to check how to get rid of the git setting.

Anyways, if you have an idea about how to test this, I can try to add something.

### CODE REVIEW

File: src/test/scala/rxscalajs/ObservableTest.scala [hunk 1]
```diff
@@ -7,11 +7,12 @@ import utest._
 
 import scala.concurrent.{ExecutionContext, Future}
 import scala.scalajs.js
+import scala.scalajs.js.|
 
 
 object ObservableTest extends TestSuite {
 
-
+  type Creator = Unit | (() => Unit)
```
Comment (LukaJCB) [inline src/test/scala/rxscalajs/ObservableTest.scala]: I think type Creator needs to explicitly be `js.Function0[Unit]`, for JS interop to work correctly

**Comment by LukaJCB** [comment]
Also you can just add a test that creates an Observable with a Dispose function and see if it's being called, without everything crashing. :)

File: src/test/scala/rxscalajs/ObservableTest.scala (hunk 1 above)
Comment (LukaJCB) [reply src/test/scala/rxscalajs/ObservableTest.scala]: Actually, you can keep it this way in the Observable.scala file, but not in the ObservableFacade.scala file. :)

**Comment by Narigo** [comment]
I've added a test after reading a bit more about µtest and hopefully fixed the typing issue you've mentioned. I was unsure if it's working, but when setting `x = false` in the `disposer` function in the test, it showed a red test, so I guess it should be okay :)

Edit: This is only a test for the `WrapperTest`. Do you need a `FacadeTest` as well?

**Comment by Narigo** [comment]
Added another test for `ObservableFacade`. Wasn't as hard as I expected it to be :)

I guess someone can make the typing stuff look better than what I did, but I'm not really getting my head around how the union type etc works yet...

**Comment by LukaJCB** [comment]
Awesome, looks good to me!

**Pull request merged**